pkg/controllers/leaderworkerset_controller.go

```diff
@@ -135,7 +137,21 @@ func (r *LeaderWorkerSetReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return ctrl.Result{}, err
 	}
 
+	if !lwsUpdated {
```

Comment:
I would also do this after the successful call to `SSAWithStatefulset` to ensure that the patch was actually successful to trigger the update

Reply:
can we also put this in an if/else statement with the creating leader event?